index.html

```diff
@@ -4208,6 +4216,122 @@ <h3 id="looping-AudioBufferSourceNode">
         </section>
       </section>
       <section>
+        <h2 id="ConstantSourceNode">
+          The ConstantSourceNode Interface
+        </h2>
+        <p>
+          This interface represents a constant audio source whose output is
+          nominally a constant value. It is useful as a constant source node in
+          general and can be used as if it were a constructible
+          <a><code>AudioParam</code></a> by automating its
+          <a><code>sourceValue</code></a> or connecting another node to it.
+        </p>
+        <p>
+          The single output of this node consists of one channel (mono).
+        </p>
+        <pre>
+  numberOfInputs  : 0
+  numberOfOutputs : 1
+        </pre>
+        <dl title=
+        "[Constructor(BaseAudioContext context, optional ConstantSourceOptions options)]interface ConstantSourceNode : AudioNode"
+        class="idl">
+          <dt>
+            readonly attribute AudioParam sourceValue
+          </dt>
+          <dd>
+            <p>
+              The constant value of the source. Its default value is 0. This
+              parameter is <a>a-rate</a>. Its nominal range is \([-\infty,
+              \infty]\).
+            </p>
+          </dd>
+          <dt>
+            attribute EventHandler onended
+          </dt>
+          <dd>
+            <p>
+              A property used to set the <code>EventHandler</code> (described
+              in <cite><a href=
+              "https://html.spec.whatwg.org/multipage/webappapis.html#eventhandler">
+              HTML</a></cite>[[!HTML]]) for the ended event that is dispatched
+              to <a><code>AudioBufferSourceNode</code></a> node types. When the
+              playback of constant is finished, an event of type
+              <code>Event</code> (described in <cite><a href=
+              "https://html.spec.whatwg.org/multipage/infrastructure.html#event">
+              HTML</a></cite> [[!HTML]]) will be dispatched to the event
+              handler.
+            </p>
+          </dd>
+          <dt>
+            void start(optional double when = 0)
+          </dt>
+          <dd>
+            <p>
+              Schedules the constant source to playback at an exact time. This
+              is defined exactly the same as for the <a href=
+              "#widl-AudioBufferSourceNode-start-void-double-when-double-offset-double-duration">
+              <code>start()</code></a> method for an
```

Comment:
I don't want to do the refactoring in this PR.  Do you want a new class before or after this PR?